uaclient/config.py

```diff
@@ -121,6 +123,26 @@ def __init__(
             self.cfg_path = get_config_path()
             self.cfg, self.invalid_keys = parse_config(self.cfg_path)
 
+        if user_config:
+            self.user_config = user_config
+        else:
+            try:
+                self.user_config = (
+                    state_files.user_config_file.read()
+                    or state_files.UserConfigData()
```

Comment:
I think we can remove the `or` clause here

Reply:
`read` could return `None`, so this `or` would give us the default data object instead of None. This makes it so we don't need to check for None in all of the property functions below

Reply:
Fair enough, the `None` case is when the file doesn't exist. My mistake